infra/token-transfer-server/src/lib/transfer.js

```diff
@@ -190,47 +195,81 @@ async function executeTransfer(transfer, transferTaskId) {
     txHash = await token.credit(transfer.toAddress, naturalAmount)
   } catch (error) {
     logger.error('Error crediting tokens', error.message)
+
     await updateTransferStatus(
       user,
       transfer,
       enums.TransferStatuses.Failed,
       TRANSFER_FAILED,
       error.message
     )
-    return { txHash: null, txStatus: enums.TransferStatuses.Failed }
+    return false
   }
 
+  logger.info(`Transfer ${transfer.id} processed with hash ${txHash}`)
+
   await transfer.update({
     status: enums.TransferStatuses.WaitingConfirmation,
     fromAddress: supplier.toLowerCase(),
     txHash
   })
 
+  return txHash
+}
+
+/**
+ * Sends a blockchain transaction to transfer tokens.
+ * @param {Transfer} transfer: DB model Transfer object
+ * @returns {Promise<>}
+ */
+async function checkBlockConfirmation(transfer) {
+  // Setup token library
+  const token = new Token(networkId)
+
   // Wait for the transaction to get confirmed.
-  const { status } = await token.waitForTxConfirmation(txHash, {
-    numBlocks: NumBlockConfirmation,
-    timeoutSec: ConfirmationTimeoutSec
+  const result = await token.txIsConfirmed(transfer.txHash, {
+    numBlocks: NumBlockConfirmation
   })
 
   let transferStatus, eventAction, failureReason
-  switch (status) {
-    case 'confirmed':
-      transferStatus = enums.TransferStatuses.Success
-      eventAction = TRANSFER_DONE
-      break
-    case 'failed':
-      transferStatus = enums.TransferStatuses.Failed
-      eventAction = TRANSFER_FAILED
-      break
-    case 'timeout':
+  if (!result) {
+    // Check if too long elapsed
+    const timedOut =
+      moment.utc(transfer.updatedAt).diff(moment.utc(), 'seconds') >
+      ConfirmationTimeoutSec
+    if (timedOut) {
       transferStatus = enums.TransferStatuses.Failed
       eventAction = TRANSFER_FAILED
       failureReason = 'Confirmation timeout'
-      break
-    default:
-      throw new Error(`Unexpected status ${status} for txHash ${txHash}`)
+      logger.warn(`Transaction timed out ${transfer.txHash}`)
+    }
+    return null
```

Comment:
Should there be a else statement here so that null is only return if it is not a timeout?

Reply:
My bad!